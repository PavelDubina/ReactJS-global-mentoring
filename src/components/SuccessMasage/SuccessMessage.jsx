import React from 'react'
import PropTypes from 'prop-types'
import { Modal } from '../Modal/Modal'
import styles from './SuccessMesage.scss'

export const SuccessMesage = ({ onClose }) => (
  <Modal onClose={onClose}>
    <div className={styles.container}>
      <div className={styles.img} />
      <h1>CONGRATULATIONS !</h1>
      <p>The movie has been added to database successfully</p>
    </div>
  </Modal>
)

SuccessMesage.propTypes = {
  onClose: PropTypes.func.isRequired,
}
