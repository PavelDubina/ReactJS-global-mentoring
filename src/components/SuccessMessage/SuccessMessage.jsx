import React from 'react'
import PropTypes from 'prop-types'
import { Modal } from '../Modal/Modal'
import styles from './SuccessMessage.scss'

export const SuccessMessage = ({ onClose }) => (
  <Modal onClose={onClose}>
    <div className={styles.container}>
      <div className={styles.img} />
      <h1>CONGRATULATIONS !</h1>
      <p>The movie has been added to database successfully</p>
    </div>
  </Modal>
)

SuccessMessage.propTypes = {
  onClose: PropTypes.func.isRequired,
}
