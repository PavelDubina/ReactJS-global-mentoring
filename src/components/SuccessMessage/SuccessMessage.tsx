import React from 'react'
import { useTranslation } from 'next-i18next'
import { Modal } from '../Modal/Modal'
import styles from '../../../styles/SuccessMessage.module.scss'

type SuccessMessageProps = {
  onClose: (isMessage?: boolean) => void
}

export const SuccessMessage: React.FC<SuccessMessageProps> = ({ onClose }) => {
  const { t } = useTranslation()
  return (
    <Modal onClose={onClose}>
      <div className={styles.container}>
        <div className={styles.img} />
        <h1>{t('success.title')}</h1>
        <p>{t('success.desc')}</p>
      </div>
    </Modal>
  )
}
