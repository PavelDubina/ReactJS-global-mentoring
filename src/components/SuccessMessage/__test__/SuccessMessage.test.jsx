import React from 'react'
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react'
import { SuccessMessage } from '../SuccessMessage'

describe('SuccessMessage', () => {
  ReactDOM.createPortal = jest.fn((element) => element)
  const getComponent = ({ onClose = jest.fn() }) => render(<SuccessMessage onClose={onClose} />)
  it('Should renders correctly', () => {
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
})
