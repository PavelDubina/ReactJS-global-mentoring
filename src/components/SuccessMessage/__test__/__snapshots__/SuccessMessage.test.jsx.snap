// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessMessage Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="modal_backdrop"
  >
    <div
      class="modal_window"
    >
      <div
        class="close_btn"
      >
        Ã—
      </div>
      <h1
        class="title"
      />
      <div
        class="container"
      >
        <div
          class="img"
        />
        <h1>
          CONGRATULATIONS !
        </h1>
        <p>
          The movie has been added to database successfully
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
